R/runCalibration.R

```diff
@@ -814,24 +888,57 @@ runCalibrationOptim <- function(path,
 #' @param xinit data frame with initial specific intangible costs
 #' @param dims character, dimensions of the optimization variable
 #' @param tcalib numeric, calibration time steps
+#' @param flow character, type of flow, either construction or renovation
 #' @param varName character, optimization variable to calculate specific costs from.
 #'   Should be one of 'x', 'xA' or 'xMin'.
+#' @param vinExists data frame of vintages that exist for each time period
+#' @param vinCalib data frame with vintages that exist in calibration periods
 #'
 #' @importFrom dplyr %>% .data across any_of group_by mutate select ungroup
 #'
 # TODO: Maybe handle case of zero values differently
-.determineSpecCost <- function(optimVar, xinit, dims, tcalib, varName = "x") {
-  xinit %>%
+.determineSpecCost <- function(optimVar, xinit, dims, tcalib, flow = c("construction", "renovation"),
+                               varName = "x", vinExists = NULL, vinCalib = NULL) {
+
+  flow <- match.arg(flow)
+
+  specCost <- xinit %>%
+    .filter(vinExists) %>%
+    replace_na(list(value = 0)) %>% # replace missing initial values with 0
     left_join(optimVar, by = dims) %>%
     group_by(across(-any_of(c("ttot", "x", "xA", "xMin", "value")))) %>%
     mutate(xProj = mean(.data[[varName]][.data[["ttot"]] %in% tcalib]),
            value = ifelse(
              .data[["ttot"]] %in% tcalib,
              .data[["value"]] + .data[[varName]],
              .data[["value"]] + .data[["xProj"]]
-           )) %>%
+           ))
+  if (flow == "renovation") {
+    if (all(grepl("all", vinCalib$vin))) {
+      vinCalib <- vinCalib %>%
+        select(-"vin") %>%
+        left_join(vinExists, by = "ttot")
+    }
+    vinCalibMax <- vinCalib %>%
+      mutate(to = as.numeric(sub(".*(\\d{4})$", "\\1", as.character(.data$vin)))) %>%
+      filter(.data$to == max(.data$to)) %>%
+      dplyr::pull(var = "vin") %>%
+      unique()
+
+    specCost <- specCost %>%
+      ungroup() %>%
+      group_by(across(-any_of(c("vin", "x", "xA", "xMin", "xProj", "value")))) %>%
```

Comment:
I thought previous grouping doesn't affect new grouping.
```suggestion
      group_by(across(-any_of(c("vin", "x", "xA", "xMin", "xProj", "value")))) %>%
```

Reply:
Ah thanks, I didn't know that.